Tasks/PublishSymbols/Helpers.ps1

```diff
@@ -276,17 +233,221 @@ function Get-SourceFilePaths {
+            if ($provider.PublicCollectionUrl.EndsWith(
+                    [Microsoft.TeamFoundation.Framework.Common.LocationServiceConstants]::CollectionLocationServiceRelativePath,
+                    [System.StringComparison]::OrdinalIgnoreCase)) {
+                $provider.PublicCollectionUrl = $provider.PublicCollectionUrl.Substring(
+                    0,
+                    $provider.PublicCollectionUrl.Length - [Microsoft.TeamFoundation.Framework.Common.LocationServiceConstants]::CollectionLocationServiceRelativePath.Length)
+            } else {
+                # TODO: SHOULD THIS REVERT TO ENV VAR?
+            }
+
+            $success = $true
+            return New-Object psobject -Property $provider
+        }
+
+        Write-Warning (Get-LocalizedString -Key 'Only TfsGit and TfsVersionControl source providers are supported for source indexing. Repository type: {0}' -ArgumentList $provider)
+        Write-Warning (Get-LocalizedString -Key 'Unable to index sources.')
+        return
+    } finally {
+        if (!$success) {
+            Invoke-DisposeSourceProvider -Provider $provider
+        }
+    }
+}
+
+function Invoke-DisposeSourceProvider {
+    [cmdletbinding()]
+    param($Provider)
+
+    if ($Provider.TfsTeamProjectCollection) {
+        Write-Verbose 'Disposing tfsTeamProjectCollection'
+        $Provider.TfsTeamProjectCollection.Dispose()
+        $Provider.TfsTeamProjectCollection = $null
+    }
+}
+
+function Invoke-IndexSources {
     [cmdletbinding()]
     param(
-        [string]$CollectionUrl = $(throw 'Missing CollectionUrl'),
-        [string]$TeamProjectId = $(throw 'Missing TeamProjectId'),
-        [string]$RepoId = $(throw 'Missing RepoId'),
-        [string]$CommitId = $(throw 'Missing CommitId'),
-        [string]$SourcesRootPath = $(throw 'Missing SourcesRootPath'),
-        [string[]]$SourceFilePaths = $(throw 'Missing SourceFilePaths')
+        [string[]]$SymbolsFilePaths = $(throw 'Missing SymbolsFilePaths'),
+        [switch]$TreatNotIndexedAsWarning
     )
 
+    # Validate at least one symbols file.
+    if (!$SymbolsFilePaths) {
+        Write-Warning (Get-LocalizedString -Key 'No files were selected for indexing.');
+        return
+    }
+
+    # Resolve location of pdbstr.exe.
+    if (!($pdbstrPath = Get-ToolPath -Name 'Pdbstr\pdbstr.exe')) {
+        throw (Get-LocalizedString -Key 'Could not find pdbstr.exe')
+    }
+
+    # Warn if spaces in the temp path.
+    if ($env:TMP.Contains(' ')) {
+        Write-Warning (Get-LocalizedString -Key 'Source files may not be indexed properly. Temp folder contains spaces in the path. To fix the issue, set the temp folder to a path without spaces.')
+        # Don't short-circuit. Just try anyway even though it will likely fail.
+    }
+
+    # For consistency with the previous implementation, set the working directory to the temp folder
+    # before calling pdbstr.exe.
+    Push-Location $env:TEMP
```

Comment:
I know this is a mismatch from the TMP var used above. It doesn't really matter. The TMP var above is consistent with the variable that New-Temporary file uses, so it's the correct one to use above. The TEMP var here is just what it directory is pushed before running the logic. I *think* it would be OK to switch this over to TMP, but I really don't want to touch it if it ain't broke. PDBSTR.exe is really sensitive and I don't want to introduce any new problems.